### PR TITLE
Feature/add boolqa config

### PR DESCRIPTION
**Pull request by vaskonov**
Add the boolqa config
https://github.com/google-research-datasets/boolean-questions

### CODE REVIEW

File: deeppavlov/dataset_readers/boolqa_reader.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+from deeppavlov.core.commands.utils import expand_path
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.data.dataset_reader import DatasetReader
+from deeppavlov.core.data.utils import download_decompress
+
+@register('boolqa_reader')
+class BoolqaReader(DatasetReader):
+    """
+    The class to read the BoolQ dataset from files. BoolQ is a question answering dataset for yes/no questions containing 15942 examples. These questions are naturally occurring ---they are generated in unprompted and unconstrained settings. Each example is a triplet of (question, passage, answer), with the title of the page as optional additional context. The text-pair classification setup is similar to existing natural language inference tasks.
+More details are available in https://arxiv.org/abs/1905.10044
```
Comment (yurakuratov) [inline deeppavlov/dataset_readers/boolqa_reader.py:28]: Could you also add reference to the Russian version of dataset?

Comment (vaskonov) [reply deeppavlov/dataset_readers/boolqa_reader.py]: done

File: deeppavlov/dataset_readers/boolqa_reader.py [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    The class to read the BoolQ dataset from files. BoolQ is a question answering dataset for yes/no questions containing 15942 examples. These questions are naturally occurring ---they are generated in unprompted and unconstrained settings. Each example is a triplet of (question, passage, answer), with the title of the page as optional additional context. The text-pair classification setup is similar to existing natural language inference tasks.
+
+More details about the English BoolQ are available in https://arxiv.org/abs/1905.10044
+https://github.com/google-research-datasets/boolean-questions
+
+The details about the Russian DaNetQA are available in 
+https://russiansuperglue.com/ru/tasks/task_info/DaNetQA
+
+The reader supports the English and the Russian variants of the dataset.
+The config example is boolqa_rubert.json.
```
Comment (IgnatovFedor) [inline deeppavlov/dataset_readers/boolqa_reader.py:36]:
```suggestion
    More details about the English BoolQ are available in https://arxiv.org/abs/1905.10044
    https://github.com/google-research-datasets/boolean-questions

    The details about the Russian DaNetQA are available in 
    https://russiansuperglue.com/ru/tasks/task_info/DaNetQA

    The reader supports English and Russian variants of the dataset.
    The config example is boolqa_rubert.json.
```

File: deeppavlov/dataset_readers/boolqa_reader.py [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+import json
+from pathlib import Path
+from typing import Dict, List, Tuple
+
+from deeppavlov.core.commands.utils import expand_path
+from deeppavlov.core.common.registry import register
+from deeppavlov.core.data.dataset_reader import DatasetReader
+from deeppavlov.core.data.utils import download_decompress
+
```
Comment (IgnatovFedor) [inline deeppavlov/dataset_readers/boolqa_reader.py:23]:
```suggestion


```

File: deeppavlov/dataset_readers/boolqa_reader.py [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+            'en': 'http://files.deeppavlov.ai/datasets/BoolQ.tar.gz',
+            'ru': 'http://files.deeppavlov.ai/datasets/DaNetQA.tar.gz'
+           }
+
+    def read(self,
+             data_path: str,
+             language: str = 'en',
+             *args, **kwargs) -> Dict[str, List[Tuple[Tuple[str, str], int]]]:
+
+        """Read the BoolQ dataset from files.
```
Comment (IgnatovFedor) [inline deeppavlov/dataset_readers/boolqa_reader.py:49]:
```suggestion
        """Reads BoolQ dataset from files.
```

File: deeppavlov/dataset_readers/boolqa_reader.py [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+
+        Args:
+            data_path: A path to a folder with dataset files.
+            language: The dataset language ('ru', 'en' are available)
+        """
+
+        if language in self.urls:
+            self.url = self.urls[language]
+        else:
+             raise RuntimeError(f'The dataset for {language} is unavailable')
```
Comment (IgnatovFedor) [inline deeppavlov/dataset_readers/boolqa_reader.py:59]:
```suggestion
            raise RuntimeError(f'The dataset for {language} is unavailable')
```

File: deeppavlov/dataset_readers/boolqa_reader.py [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+
+        data_path = expand_path(data_path)
+        if not data_path.exists():
+            data_path.mkdir(parents=True)
+
+        download_decompress(self.url, data_path)
+        dataset = {}
+
+        for filename in ['train.jsonl', 'valid.jsonl']:
+           dataset[filename.split('.')[0]] = self._build_data(language, data_path / filename)
```
Comment (IgnatovFedor) [inline deeppavlov/dataset_readers/boolqa_reader.py:69]:
```suggestion
            dataset[filename.split('.')[0]] = self._build_data(language, data_path / filename)
```

File: deeppavlov/dataset_readers/boolqa_reader.py [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+    @staticmethod
+    def _build_data(ln: str, data_path: Path) -> List[Tuple[Tuple[str, str], int]]:
+
+        data = {}
+        with open(data_path, 'r') as f:
+            for line in f:
+                jline = json.loads(line)
+                if ln == 'ru':
+                    if 'label' in jline:
+                        data[tuple([jline['question'], jline['passage']])] = int(jline['label'])
```
Comment (IgnatovFedor) [inline deeppavlov/dataset_readers/boolqa_reader.py:82]:
```suggestion
                        data[jline['question'], jline['passage']] = int(jline['label'])
```

File: deeppavlov/dataset_readers/boolqa_reader.py [hunk 8]
```diff
@@ -0,0 +1,87 @@
...
+                if ln == 'en':
+                    if 'answer' in jline:
+                        data[tuple([jline['question'], jline['passage']])] = int(jline['answer'])
```
Comment (IgnatovFedor) [inline deeppavlov/dataset_readers/boolqa_reader.py:85]:
```suggestion
                        data[jline['question'], jline['passage']] = int(jline['answer'])
```

File: deeppavlov/dataset_readers/boolqa_reader.py [hunk 9]
```diff
@@ -0,0 +1,87 @@
...
+             data_path: str,
+             language: str = 'en',
+             *args, **kwargs) -> Dict[str, List[Tuple[Tuple[str, str], int]]]:
+
+        """Read the BoolQ dataset from files.
+
+        Args:
+            data_path: A path to a folder with dataset files.
+            language: The dataset language ('ru', 'en' are available)
+        """
```
Comment (IgnatovFedor) [inline deeppavlov/dataset_readers/boolqa_reader.py:54]: Add `Returns:` to description

File: deeppavlov/core/common/registry.json [hunk 10]
```diff
@@ -187,4 +188,4 @@
   "wiki_parser_online": "deeppavlov.models.kbqa.wiki_parser_online:WikiParserOnline",
   "wiki_sqlite_vocab": "deeppavlov.vocabs.wiki_sqlite:WikiSQLiteVocab",
   "wikitionary_100K_vocab": "deeppavlov.vocabs.typos:Wiki100KDictionary"
-}
+}
```
Comment (IgnatovFedor) [inline deeppavlov/core/common/registry.json:191]: add newline to the end of file

File: deeppavlov/dataset_readers/boolqa_reader.py (hunk 9 above)
Comment (vaskonov) [reply deeppavlov/dataset_readers/boolqa_reader.py]: done

File: deeppavlov/core/common/registry.json (hunk 10 above)
Comment (vaskonov) [reply deeppavlov/core/common/registry.json:191]: The file generated automatically

**Pull request merged**